civicrm.php

```diff
@@ -1451,7 +1451,7 @@ public function is_page_request() {
      * unnecessary?
      */
     if (CRM_Utils_Array::value('HTTP_X_REQUESTED_WITH', $_SERVER) === 'XMLHttpRequest'
-        || ($argdata['args'][0] === 'civicrm' && in_array($argdata['args'][1], ['ajax', 'file']))
+        || ($argdata['args'][0] === 'civicrm' && in_array($argdata['args'][1], ['ajax', 'file', 'asset']))
```

Comment:
Wow, what a beautifully awful hack.

Long-term, seems like we ought to include this sort of thing in our router metadata instead of *guessing* what kind of route it is based on string-matching.

Reply:
Goes waaaay back [to 4.4](https://github.com/civicrm/civicrm-wordpress/pull/42).

And yes, some other way would definitely be better 😆👍 

FWIW I had a thought on how to make it slightly less hacky in the short term... I'll open a PR where we can discuss further.

Reply:
@colemanw We seem to have had almost the [same conversation](https://github.com/civicrm/civicrm-wordpress/pull/40#issuecomment-59153582) 10 years ago 👻